### DISCUSSION TITLE
Apostrophe and citation keys

### DISCUSSION THREAD

**Discussion by xracoonx** [retorquere/zotero-better-bibtex#2591] [Q&A]
While the citation key is "zotero.clean" on both macOS and Windows, I get different citation keys, e.g. for "Joe's" as title, I will get "joes" as title on macOS but "joe" as title on Windows 7. Obviously, this leads to problems. Is this a bug? Or am I doing something wrong?

**Comment by retorquere** [comment]
No, that's not something that should differ. What do you get if you leave off `.clean`?

**Reply by xracoonx** [reply]
What is expected to happen? On Windows, it changes nothing, i.e. there is no "s". I cannot test at the moment whether the "s" disappears on macOS. Will do when I get to a Mac.

**Reply by retorquere** [reply]
I needed to know whether the difference came from my `clean` filter or from the `zotero` function. The `clean` filter is my doing, so if that's doing something wrong, I could fix it. The `zotero` filter is literally lifted from the Zotero BibTeX translator, so any varying behavior would have to be addressed there.

**Reply by retorquere** [reply]
It's one of the reasons why I used to suggest in the first-run wizard of BBT that you should only use `zotero` when you have existing work where you used Zotero without BBT, and you want to have a decent chance<sup>1</sup> that you get the same keys when you start using BBT. Zotero also doesn't guarantee you get the same keys if you export the same item twice together with differing other items. If you don't have such existing work, I would not use the `zotero` function.

<sup>1</sup> you are guaranteed the same keys if no two items generate the same `zotero` key. If two items generate the same key, BBT will postfix one of them, and keep them stable that way. Zotero will *sometimes* postfix one of them in this scenario, and when it does postfix one of them, it is not deterministic which of them gets the postfix. But apparently this stable non-clashing behavior is also platform-dependent. One more reason to not use the `zotero` function.

**Reply by xracoonx** [reply]
I checked on macOS now. But it produces the "s" with and without ".clean". So, I used the manual key 

`auth.lower + "_" + shorttitle(1,0).lower + "_" + year`

which seems to reproduce the `zotero` and `zotero.clean` key. (At least on Windows.)

**Reply by retorquere** [reply]
that should be stable across platforms. If not, that's a bug you can open an issue for. It's just the `zotero` filter that I specifically cannot vouch for, as it is not my code, and intentionally whatever-zotero-does.

**Reply by xracoonx** [reply]
That might be a misunderstanding:

`auth.lower + "_" + shorttitle(1,0).lower + "_" + year`

produces the same on macOS and Windows. It is just that it is the same as `zotero`(`.clean`) only on Windows. On macOS there is an extra "s" with with the latter.

**Reply by retorquere** [reply]
Then please open an issue for that, if you send a debug log (with `-refs-` in it) I will have a look.
